### PR TITLE
Copter: Factor out common method for guided and auto takeoff

### PR DESCRIPTION
**Pull request by peterbarker**
Saves a couple of hundred bytes.

### CODE REVIEW

File: ArduCopter/mode_guided.cpp [hunk 1]
```diff
@@ -366,6 +366,11 @@ void Copter::ModeGuided::run()
 // guided_takeoff_run - takeoff in guided mode
 //      called by guided_run at 100hz or more
 void Copter::ModeGuided::takeoff_run()
+{
+    auto_takeoff_run();
+}
+
+void Copter::Mode::auto_takeoff_run()
```
Comment (khancyr) [inline ArduCopter/mode_guided.cpp]: shouldn't this be move in takeoff.cpp ?

Comment (peterbarker) [reply ArduCopter/mode_guided.cpp]: We should probably split into `user_takeoff.cpp` and `auto_takeoff.cpp`, I think.  But not as part of this PR :-)

**Comment by peterbarker** [comment]
On Mon, 2 Jul 2018, Francisco Ferreira wrote:

> @peterbarker More and more I think we need a shared state between modes, the static variable here is another hint of it.

In this particular case we could create something similar to the "user 
takeoff" object (Mode::_TakeOff) to hold that state.  These: 
https://github.com/ArduPilot/ardupilot/pull/8740/files#diff-973794592185a0184d85bf6730b44664R170 
would become methods and a state variable.

So Mode::_AutoTakeOff::run(void) etc etc

**Comment by khancyr** [comment]
@rmackay9 ping ! 
tested and validated on SITL.

**Comment by peterbarker** [comment]
Randy said this was OK in #9010

**Pull request merged**